### DISCUSSION TITLE
something keeps removing old history entries

### DISCUSSION THREAD

**Discussion by terminaldweller** [ohmyzsh/ohmyzsh#12312] [Help]
like the title says, something keeps removing older zsh history entries.
what i have set in my zshrc:

```zsh
export HISTSIZE=50000
export SAVEHIST=50000
setopt APPEND_HISTORY
setopt EXTENDED_HISTORY
setopt HIST_FIND_NO_DUPS
setopt HIST_IGNORE_ALL_DUPS
setopt HIST_IGNORE_SPACE
setopt HIST_NO_STORE
setopt HIST_REDUCE_BLANKS
setopt HIST_SAVE_NO_DUPS
setopt HIST_EXPIRE_DUPS_FIRST
setopt HIST_SAVE_BY_COPY
setopt HIST_FIND_NO_DUPS
setopt HIST_VERIFY
setopt SHARE_HISTORY
setopt AUTO_CD
setopt AUTO_PUSHD
setopt NO_BEEP
setopt INTERACTIVE_COMMENTS
setopt PROMPT_SUBST
unsetopt FLOW_CONTROL
setopt LONG_LIST_JOBS
setopt +o nomatch
export HISTFILE=~/.zsh_history
setopt LIST_PACKED
setopt INC_APPEND_HISTORY
```

any ideas what can do this?

**Comment by rafaelpagliari** [comment]
Hello, I adjusted it like this:

An update was applied which caused this issue, I solved it by:
editing the file path=.oh-my-zsh/lib/history.zsh

and reverting the changes made in this commit: https://github.com/ohmyzsh/ohmyzsh/commit/a8f7f9f707ca5fb9ac7bedb088748b09db2fd3bd

**Reply by IgaoWolf** [reply]
Nice @rafaelpagliari, it`s work for me. Tks

**Reply by terminaldweller** [reply]
this, unfortunately is unrelated to my issue since I never was using `history` to check my history file contents. I was looking at the history file itself. Also, at the time of making this discussion thread, I did not have the mentioned bugfix.

**Reply by mcornella** [reply]
The issue reported about using `history` and not showing all entries is now fixed.

**Comment by mcornella** [comment]
@terminaldweller would be helpful to know specifically which entries are being deleted, especifically of note would be to know how many commands are in the history, since `HISTSIZE` and `SAVEHIST` will have an impact on that.

See https://unix.stackexchange.com/questions/273861/unlimited-history-in-zsh for more context on this, and https://zsh.sourceforge.io/Doc/Release/Parameters.html for the reference of what those variables specify.

**Reply by terminaldweller** [reply]
right now i have around 2374 entries. The oldest entries get removed. so the ones from the top of the hist file get removed.
also thanks for the links. I'll take a look at the links.

**Reply by terminaldweller** [reply]
I think I figured it out.
I just ran `omz update` and lost about 500ish older history entries.

**Reply by jaysonwcs** [reply]
This is happening to me every time I update oh-my-zsh. Did you find where the problem is? Is there any workaround?
